<?xml version="1.0"?>
<launch>
    <arg name="robot_name" default="ropod_001"/>
    <arg name="debug_mode" default="false"/>

    <node name="ropod_com_mediator" pkg="ropod_com_mediator" type="ropod_com_mediator" 
        output="screen" args="ropod_com_mediator robot_name=$(arg robot_name) debug_mode=$(arg debug_mode)">
        <param name="zyreGroupName" type="str" value="ROPOD"/>

        <remap from="~robot_pose" to="/ropod/pose"/>
        <remap from="~robot_subarea" to="/ropod/subarea" />
	    <remap from="~command" to="/ropod/remote_command" />
    </node>

    <node name="robot_pose_publisher" pkg="ropod_com_mediator" type="robot_pose_publisher.py" output="screen">
        <param name="reference_frame" type="str" value="map"/>
        <param name="target_frame" type="str" value="ropod/base_link"/>
        <remap from="~pose" to="/ropod/pose"/>
    </node>
    <node name="robot_command_processor" pkg="ropod_com_mediator" type="robot_command_processor.py" output="screen">
        <remap from="~command" to="/ropod/remote_command"/>
        <remap from="~joypad" to="/joy"/>
        <remap from="~nav_pause" to="/ropod/napoleon/pause"/>
    </node>
</launch>
